flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/SpeculativeScheduler.java

```diff
@@ -149,6 +159,16 @@ public SpeculativeScheduler(
         this.blocklistOperations = checkNotNull(blocklistOperations);
 
         this.slowTaskDetector = new ExecutionTimeBasedSlowTaskDetector(jobMasterConfiguration);
+
+        this.numEffectiveSpeculativeExecutionsCounter = new SimpleCounter();
+        this.registerMetrics(jobManagerJobMetricGroup);
```

Comment:
I think it would be better to move the registration to `startSchedulingInternal`